accessible/workspace_treeview.component.js

```diff
@@ -0,0 +1,212 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Angular2 Component that details how Blockly.Block's are
+ * rendered in the workspace in AccessibleBlockly. Also handles any
+ * interactions with the blocks.
+ * @author madeeha@google.com (Madeeha Ghori)
+ */
+blocklyApp.WorkspaceTreeView = ng.core
+  .Component({
+    selector: 'tree-view',
+    template: `
+      <li #parentList [id]="idMap['parentList']" role="treeitem" class="blocklyHasChildren" 
+          [attr.aria-labelledBy]="utilsService.generateAriaLabelledByAttr('blockly-block-summary', idMap['blockSummary'])" 
+          [attr.aria-level]="level" aria-selected=false>
+        {{checkParentList(parentList)}}
+        <label [id]="idMap['blockSummary']">{{block.toString()}}</label>
+        <ol role="group"  [attr.aria-level]="level+1">
+          <li [id]="idMap['listItem']" class="blocklyHasChildren" role="treeitem" 
+              [attr.aria-labelledBy]="utilsService.generateAriaLabelledByAttr('blockly-block-menu', idMap['blockSummary'])" 
+              [attr.aria-level]="level+1" aria-selected=false>
+            <label [id]="idMap['label']">{{stringMap['BLOCK_ACTION_LIST']}}</label>
+            <ol role="group"  [attr.aria-level]="level+2">
+              <li [id]="idMap['cutListItem']" role="treeitem" 
+                  [attr.aria-labelledBy]="utilsService.generateAriaLabelledByAttr(idMap['cutButton'], 'blockly-button')" 
+                  [attr.aria-level]="level+2" aria-selected=false>
+                <button [id]="idMap['cutButton']" (click)="clipboardService.cut(block)">{{stringMap['CUT_BLOCK']}}</button>
+              </li>
+              <li [id]="idMap['copyListItem']" role="treeitem" 
+                  [attr.aria-labelledBy]="utilsService.generateAriaLabelledByAttr(idMap['copyButton'], 'blockly-button')" 
+                  [attr.aria-level]="level+2" aria-selected=false>
+                <button [id]="idMap['copyButton']" (click)="clipboardService.copy(block, true)">{{stringMap['COPY_BLOCK']}}</button>
+              </li>
+              <li [id]="idMap['pasteBelow']" role="treeitem" 
+                  [attr.aria-labelledBy]="utilsService.generateAriaLabelledByAttr(idMap['pasteBelowButton'], 'blockly-button', (noNextConnectionHTMLText(block)||clipboardCompatibilityHTMLText(block.nextConnection)))"  
+                  [attr.aria-level]="level+2" aria-selected=false>
+                <button [id]="idMap['pasteBelowButton']" (click)="clipboardService.pasteFromClipboard(block.nextConnection)" 
+                    [disabled]="noNextConnectionHTMLText(block)" [disabled]="clipboardCompatibilityHTMLText(block.nextConnection)">{{stringMap['PASTE_BELOW']}}</button>
```

Comment:
Is the intend to have two disabled attrs for the same tag? One will be ignored, I think.